src/bin/pgcopydb/env_utils.c

```diff
@@ -215,3 +218,104 @@ get_env_pgdata_or_exit(char *pgdata)
 			  "or from --pgdata");
 	exit(EXIT_CODE_BAD_ARGS);
 }
+
+
+/*
+ * get_env_using_parsers iterates over the parsers array and calls
+ * get_env_using_parser for each parser.
+ */
+bool
+get_env_using_parsers(EnvParserArray *parsers)
+{
+	int errors = 0;
+
+	for (int i = 0; i < parsers->count; i++)
+	{
+		if (!get_env_using_parser(&parsers->array[i]))
+		{
+			++errors;
+		}
+	}
+
+	return errors == 0;
+}
+
+
+/*
+ * get_env_using_parser checks if the environment variable exists and if it does
+ * it parses it and stores it in the target. If the environment variable is not
+ * set, the target is not modified. If the environment variable is set but the
+ * parsing fails, an error is logged and the errors counter is incremented.
+ */
+bool
+get_env_using_parser(EnvParser *parser)
+{
+	if (!env_exists(parser->envname))
+	{
+		return true;
+	}
+
+	switch (parser->type)
+	{
+		case ENV_TYPE_INT:
+		{
+			char envValue[BUFSIZE] = { 0 };
+			if (!get_env_copy(parser->envname, envValue, sizeof(envValue)))
+			{
+				/* errors have already been logged */
+				return false;
+			}
+			else if (!stringToInt(envValue, (int *) parser->target) ||
+					 (parser->lowerBounded &&
+					  *((int *) parser->target) < parser->minValue) ||
+					 (parser->upperBounded &&
+					  *((int *) parser->target) > parser->maxValue))
+			{
+				log_fatal("Failed to parse \"%s\": \"%s\"", parser->envname,
+						  envValue);
+				return false;
+			}
```

Comment:
I think we need a specific error message for values out of bound, so that the user has a chance to fix it.

Reply:
Good suggestion.

I needed a way to manipulate character arrays, and I decided to use `PQExpBuffer` to generate the error message. If you can think of a better way to implement this, please share it with me.